### PR TITLE
Add check for generating manifests and code

### PR DESCRIPTION
**Pull request by ChenYi015**
## Purpose of this PR

Provide a clear and concise description of the changes. Explain the motivation behind these changes and link to relevant issues or discussions.

**Proposed changes:**
- Add CI check for generating CustomResourceDefinition, RBAC and WebhookConfiguration manifests.
- Add CI check for generating code containing DeepCopy, DeepCopyInto, and DeepCopyObject method implementations.

## Change Category
Indicate the type of change by marking the applicable boxes:

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] Feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that could affect existing functionality)
- [ ] Documentation update

## Checklist
Before submitting your PR, please review the following:

- [x] I have conducted a self-review of my own code.
- [ ] I have updated documentation accordingly.
- [] I have added tests that prove my changes are effective or that my feature works.
- [x] Existing unit tests pass locally with my changes.

### CODE REVIEW

File: .github/workflows/integration.yaml [hunk 1]
```diff
@@ -145,6 +145,27 @@ jobs:
...
+          fi
+
+      - name: Generate code
+        run: |
+          make generate
+          if ! git diff --quiet; then
+            echo "Need to re-run 'make generate' and commit the changes."
+            git diff
+            false
+          fi
```
Comment (ImpSy) [inline .github/workflows/integration.yaml:164]: Don't you think we should test this in the `code-check` job instead of `build-helm-chart` ?

Comment (ChenYi015) [reply .github/workflows/integration.yaml]: Done.

**Review by ImpSy — APPROVED** [review]
/lgtm

**Pull request merged**